src/conventions/h-group/clue-interpretation/connecting-cards.js

```diff
@@ -55,9 +55,15 @@ export function find_known_connecting(game, giver, identity, ignoreOrders = [],
 
 			// Remove inferences that will be proven false (i.e. after someone plays the card with such identity)
 			// Except the giver, who can't eliminate from their own hand
-			if (giver !== playerIndex)
+			if (giver !== playerIndex) {
 				card.inferred = card.inferred.subtract(card.inferred.filter(inf => inf.playedBefore(identity)));
 
+				// If a waiting connection will reveal this card, assume it will be known in time.
+				const connection = common.waiting_connections.find(conn => conn.focused_card.order == card.order);
```

Comment:
I think we should explicitly exclude symmetric connections. I'm not sure if it will make a difference because we won't find connections that don't exist, but I think it's better to be safe.

Also, I think we should exclude connecting on us in this way, because we can't tell if the waiting connection is correct or not. If we are promised some card, then it should be caught by play links as a 'playable' connection.

Reply:
Done. I've added both these conditions and tests to ensure that the case is handled correctly both from the finessed player and the player who has the 3's.